tests/rewriting/composable_rewriting/immutable_ir/test_immutable_ir.py

```diff
@@ -0,0 +1,83 @@
+import pytest
```

Comment:
nit: Are we sure we want these tests in this subfolder hierarchy?

Reply:
I am mirroring the hierarchy of the project in the tests. We could also go for a flatter hierarchy if we prefer. 
I have no strong opinion on this. Only a desire to complete this PR

Reply:
What other hierarchy would you be proposing?
I feel that this one is fine. We could remove maybe the last folder, but otherwise I don't see how this would be a problem?

Reply:
good then!